blocksuite/framework/block-std/src/gfx/viewport.ts

```diff
@@ -60,7 +60,12 @@ export class Viewport {
 
   viewportMoved = new Slot<IVec>();
 
-  viewportUpdated = new Slot<{ zoom: number; center: IVec }>();
+  viewportUpdated = new Slot<{
+    zoom: number;
+    center: IVec;
+    zooming: boolean;
```

Comment:
You're storing states in this slot. These states should be replaced with some signals.